### PR TITLE
Add request response

### PR DESCRIPTION
**Pull request by KairaMiu**
Added support of MQTT 5 feature Request - Response Pattern

### CODE REVIEW

File: alm-mqtt-module/pkg/client/client.go [hunk 1]
```diff
@@ -159,3 +159,48 @@ func (c *Client) PublishOnMqttTopic(topic string, payload []byte) error {
...
+	codec, err := goavro.NewCodec(schema.ReqResRequest)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	bytes, err := avro.Writer(msg, codec)
+	if err != nil {
+		return []byte{}, err
+	}
+	response, err := c.nats.Request(fmt.Sprintf("%s.request-response", c.target), bytes, 20*time.Second)
```
Comment (KlausPopp) [inline alm-mqtt-module/pkg/client/client.go:178]: timeout should be user's timeout plus some extra

File: alm-mqtt-module/pkg/schema/avro_schemas/reqResRequest.avsc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            "type": "string"
+        },
+        {
+            "name": "payload",
+            "type": "bytes"
+        },
+        {
+            "name": "timeout",
+            "type": {
+                "doc": "Sets the timeout for a the request repsose",
```
Comment (KlausPopp) [inline alm-mqtt-module/pkg/schema/avro_schemas/reqResRequest.avsc:17]: Remove "sets the" ("timeout for a the request repsose" is sufficient"

**Review by KlausPopp — CHANGES_REQUESTED** [review]

File: alm-mqtt-module/example/requestReply/requestReply.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
+/*
+Copyright © 2021 edgefarm.io
```
[the remark this answers is not in the document]
Comment (KairaMiu) [reply alm-mqtt-module/example/requestReply/requestReply.go:2]: Done

File: alm-mqtt-module/example/requestReply/requestReply.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		type message struct {
+			Counter int `json:"counter"`
+		}
+		msg := message{
+			Counter: counter,
+		}
+
+		b, _ := json.Marshal(msg)
+		fmt.Printf("Sending message: %s\n", b)
+		res, err := client.RequestReply("/pis/cmd/state", b, int32(5000))
```
[the remark this answers is not in the document]
Comment (KairaMiu) [reply alm-mqtt-module/example/requestReply/requestReply.go:51]: Done

File: alm-mqtt-module/pkg/client/client.go (hunk 1 above)
Comment (KairaMiu) [reply alm-mqtt-module/pkg/client/client.go:178]: Changed to `2 * timeout`

File: alm-mqtt-module/pkg/schema/avro_schemas/reqResRequest.avsc (hunk 2 above)
Comment (KairaMiu) [reply alm-mqtt-module/pkg/schema/avro_schemas/reqResRequest.avsc:17]: Ok

**Review by KlausPopp — APPROVED** [review]

**Pull request merged**